cli/operator/node.go

```diff
@@ -113,9 +111,19 @@ var StartNodeCmd = &cobra.Command{
 
 		logger.Info(fmt.Sprintf("starting %v", commons.GetBuildData()))
 
-		metricsReporter := metricsreporter.New(
-			metricsreporter.WithLogger(logger),
-		)
+		observabilityShutdown, err := observability.Initialize(
+			cmd.Parent().Short,
+			cmd.Parent().Version,
+			observability.WithMetrics())
+		if err != nil {
+			logger.Fatal("could not initialize observability configuration", zap.Error(err))
+		}
+
+		defer func() {
+			if err = observabilityShutdown(cmd.Context()); err != nil {
+				logger.Error("could not shutdown observability object", zap.Error(err))
+			}
+		}()
```

Comment:
Can `observabilityShutdown` be `nil` ? I suspect we might get "nil-pointer dereference" here (since we don't return on non-nil error above)

Reply:
That’s a good catch. There is a [conditional assignment](https://github.com/ssvlabs/ssv/blob/otel-metrics/observability/observability.go#L48) of the shutdown function variable, and the default value is `nil`.

I’m not a big fan of named return values, as they can lead to scenarios like this. However, in this case, it does serve as documentation. An alternative could be to declare a new type to address the "documentation issue", but that might be overkill.
I will fix the issue by assigning a default value. Wdyt?

Reply:
> I’m not a big fan of named return values, as they can lead to scenarios like this. However, in this case, it does serve as documentation.

Agree with all that,

can you elaborate why we even need `once.Do` in `Initialize` ? I think it overcomplicates things

Reply:
> can you elaborate why we even need once.Do in Initialize (I don't see how it could run 2+ times) ? 

It's a guard from the undesired consequences if the client mistakenly calls `Initialize` function more than one time. There are many ways of implementing it, like declaring package level variable (`isInitialized` bool) and ignore the calls if it was already initialized or declaring `init()` function which runs when package is loaded and only once (although would not work for us, since there are objects in return) and such. 
It isn't mandatory, but I think it isn't bad that some functionality acts defensively, especially functionality that is exposed to other packages.

Reply:
Personally, I don't think it makes much sense in this particular case to "try defend" like that - it would be a bug to call `Initialize` 2+ times just same as if we were to call many other functions of `.Start` / `.Init` / ... kind (and we don't wrap those with `once.Do`)

Reply:
> I see. I do not have strong opinion here. Considering we do not follow this pattern in other packages, do you think it would be better to remove it?

Yeah, I would remove `once.Do` both - to keep it simple, and to be consistent with the other code we have

Reply:
removed